libres/tests/analysis/test_update.cpp

```diff
@@ -200,6 +207,193 @@ SCENARIO("Running analysis update without row scaling on linear model",
                 REQUIRE(d_posterior_ml[n_sd - 1] < eps);
             }
         }
+
+        WHEN("Row scaling factor is 0 for both parameters") {
+            row_scaling_type *row_scaling = row_scaling_alloc();
+            for (int row = 0; row < nparam; row++)
+                row_scaling_iset(row_scaling, row, 0.0);
+
+            matrix_type *A_with_scaling = matrix_alloc_copy(A);
+
+            std::pair pair{A_with_scaling, row_scaling};
+            std::vector<std::pair<matrix_type *, const row_scaling_type *>>
+                parameters{pair};
+
+            for (int iobs = 0; iobs < obs_size; iobs++) {
+                obs_block_iset(ob, iobs, measurements[iobs], 1.0);
+            }
+            matrix_type *E = obs_data_allocE(obs_data, rng, ens_size);
+
+            analysis::run_analysis_update_with_rowscaling(
+                enkf_module, ens_mask, meas_data, obs_data, rng, E, parameters);
+
+            THEN("Updated parameter matrix should equal prior parameter "
+                 "matrix") {
+                REQUIRE(matrix_similar(A, A_with_scaling, 1e-12));
```

Comment:
This is actually one of few situations where you could use `matrix_equal()` for bitwise comparison.

Reply:
> > Seems like the distance between posterior and ML-estimate when obs_std is very low does not approach 0 with row scaling, i.e., this test fails:
> 
> I am not fully up to speed on what fails here, but mind that a row scaling value < 1 can loosely speaking be seen as an inflated observation std, i.e. the "effective std" is not "very low".

Thanks, I think that makes sense.

Reply:
> > Seems like the distance between posterior and ML-estimate when obs_std is very low does not approach 0 with row scaling, i.e., this test fails:
> 
> I am not fully up to speed on what fails here, but mind that a row scaling value < 1 can loosely speaking be seen as an inflated observation std, i.e. the "effective std" is not "very low".

Thanks, I think that makes sense.